src/select.rs

```diff
@@ -93,21 +95,28 @@ impl<'a> TElement<'a> for &'a ElementData {
 }
 
 impl<'a> Node<'a> {
-    pub fn select(&'a self, css_str: &str) -> Result<FilterNodes<Descendants<'a>>, ()> {
+    pub fn select(&'a self, css_str: &str) -> Result<FilterNodes<Descendants<'a>, Vec<Selector>>, ()> {
         let selectors = try!(parser::parse_author_origin_selector_list_from_str(css_str));
         Ok(FilterNodes{
             iter: self.descendants(),
             filter: selectors,
         })
     }
+
+    pub fn text_iter(&'a self) -> FilterNodes<Siblings<'a>, TextFilter> {
```

Comment:
Would it be ok, to leave FilterNodes generic? I think it was a mistake to not make them generic. They could be reused to write something like [NodeFilters](http://www.w3.org/TR/DOM-Level-2-Traversal-Range/traversal.html#Traversal-NodeFilter).

Or I could keep a `FilterNodes{ iter: I, pred: P}` and add SelectorNodes{ iter: I, selectors: Vec<Selector>}`.